profiles/common.go

```diff
@@ -78,16 +78,16 @@ func GetResourceIds(ctx context.Context, targetProfile TargetProfile, now time.T
 	return project, instance, dbName, err
 }
 
-func GetSQLConnectionStr(sourceProfile SourceProfile) string {
+func GetSQLConnectionStr(sourceProfile *SourceProfile) string {
```

Comment:
might make sense to pass the value rather than the reference everywhere, to safeguard the attributes from change.